shiva/src/main/scala/nl/lumc/sasc/biopet/pipelines/shiva/ShivaSvCallingReport.scala

```diff
@@ -0,0 +1,110 @@
+
+    for (sampleName <- sampleNames) {
+      val sampleCounts: Map[String, Any] = Await.result(summary.getStat(runId, PipelineName("shivasvcalling"), ModuleName("parse_sv_vcf"), SampleName(sampleName)), Duration.Inf).get
+      for ((svType, counts) <- sampleCounts.collect({ case (k, v: List[_]) => (k, v.toArray[Any]) })) {
+        val elem: Tuple2[String, Array[Long]] = (sampleName, counts.collect({ case x: Long => x }))
+        svType match {
+          case "DEL" => delCounts += elem
+          case "INS" => insCounts += elem
+          case "DUP" => dupCounts += elem
+          case "INV" => invCounts += elem
+        }
+      }
+    }
+
+    var result: Map[String, Map[String, Array[Long]]] = Map()
+    if (delCounts.exists(elem => (elem._2.sum > 0))) result = Map("DEL" -> delCounts)
+    if (insCounts.exists(elem => (elem._2.sum > 0))) result += ("INS" -> insCounts)
+    if (dupCounts.exists(elem => (elem._2.sum > 0))) result += ("DUP" -> dupCounts)
+    if (invCounts.exists(elem => (elem._2.sum > 0))) result += ("INV" -> invCounts)
+    result
+  }
+
+  def parseSummaryForTranslocations(summary: SummaryDb, runId: Int, sampleNames: Seq[String]): Map[String, Long] = {
+    var traCounts: Map[String, Long] = Map()
+    for (sampleName <- sampleNames) {
+      val counts: Map[String, Any] = Await.result(summary.getStat(runId, PipelineName("shivasvcalling"), ModuleName("parse_sv_vcf"), SampleName(sampleName)), Duration.Inf).get
+      traCounts += (sampleName -> counts.get("TRA").get.asInstanceOf[Long])
+    }
+    if (traCounts.exists(elem => elem._2 > 0)) traCounts else Map.empty
+  }
+
+  def writeTsvFiles(sampleNames: Seq[String], counts: Map[String, Map[String, Array[Long]]], svTypes: List[SvTypeForReport], outFileAllTypes: String, outDir: File): Unit = {
+
+    val tsvWriter = new PrintWriter(new File(outDir, outFileAllTypes))
+    tsvWriter.print("sv_type\tsample")
+    histogramText.foreach(bin => tsvWriter.print("\t" + bin))
+    tsvWriter.println()
+
+    for (sv <- svTypes) {
+      val countsForSvType: Map[String, Array[Long]] = counts.get(sv.svType).get
+
+      writeTsvFileForSvType(sv, countsForSvType, sampleNames, outDir)
+
+      for (sampleName <- sampleNames) {
+        tsvWriter.print(sv.svType + "\t" + sampleName + "\t")
+        tsvWriter.println(countsForSvType.get(sampleName).get.mkString("\t"))
+      }
+    }
+    tsvWriter.close()
+  }
+
+  def writeTsvFileForSvType(svType: SvTypeForReport, counts: Map[String, Array[Long]], sampleNames: Seq[String], outDir: File): Unit = {
+    val tsvWriter = new PrintWriter(new File(outDir, svType.tsvFileName))
+
+    tsvWriter.print("histogramBin")
+    sampleNames.foreach(sampleName => tsvWriter.print("\t" + sampleName))
+    tsvWriter.println()
+
+    for (i <- histogramPlotTicks.indices) {
+      tsvWriter.print(histogramPlotTicks(i))
+      sampleNames.foreach(sampleName => tsvWriter.print("\t" + counts.get(sampleName).get(i)))
+      tsvWriter.println()
+    }
```

Comment:
You can also iterate over `histogramPlotTicks` while getting each item's index. Something like:

```
histogramPlotTicks.zipWithIndex.foreach {
    case (item, index) => {
        tsvWriter.print(item)
        sampleNames.foreach(sampleName => tsvWriter.print("\t" + counts.get(sampleName).get(index))
        tsvWriter.println()
    }
}
```